test/src/main/java/org/springframework/security/test/context/support/ReactorContextTestExecutionListener.java

```diff
@@ -54,7 +54,8 @@ private static TestExecutionListener createDelegate() {
 	private static class DelegateTestExecutionListener extends AbstractTestExecutionListener {
 		@Override
 		public void beforeTestMethod(TestContext testContext) throws Exception {
-			Hooks.onLastOperator(Operators.lift((s, sub) -> new SecuritySubContext<>(sub)));
+			Authentication authentication = TestSecurityContextHolder.getContext().getAuthentication();
```

Comment:
Instead of obtaining the Authentication, please use the SecurityContext so that the original SecurityContext is used (this ensures that if a custom SecurityContext was established it is used later on as well)

Reply:
Please also add a test for custom SecurityContext being established

Reply:
Sure, give me an hour for that

Reply:
Regarding the second comment, do you mean the case when `beforeTestMethod` is called on another thread?

Reply:
@rwinch

Reply:
Or you mean the end to end test case? Because I have provided a case when `beforeTestMethod` is executed on main thread and verification is being executed on ForJoinPool thread